File: java/3099. Harshad Number.java
class Solution {
    public int sumOfTheDigitsOfHarshadNumber(int x) {
        int sum = getSum(x);
        if (x % sum == 0) return sum;
        return -1;
    }
    private int getSum(int x) {
        int sum = 0;
        while (x != 0) {
            sum += x % 10;
            x /= 10;
        }
        return sum;
    }
}

File: go/3099. Harshad Number.go
func sumOfTheDigitsOfHarshadNumber(x int) int {
	getSum := func(x int) int {
		sum := 0
		for x > 0 {
			sum += x % 10
			x /= 10
		}
		return sum
	}
	sum := getSum(x)
	if x%sum == 0 {
		return sum
	}
	return -1
}